### ISSUE TITLE
[selectors] Shouldn't there be a pseudo selector for `:unchecked` inputs?

### ISSUE THREAD

**Issue by bluemoehre** [w3c/csswg-drafts#10746]
Currently the `:checked` and `:indeterminate` pseudo-classes can apply to a `<input type="checkbox" />`:
https://drafts.csswg.org/selectors/#checked
https://drafts.csswg.org/selectors/#indeterminate

But to apply styles for an *unchecked* checkbox, you must use `:not(:checked)`.

Regarding other selectors there is always a negative variant:
https://drafts.csswg.org/selectors/#enabled-pseudo
https://drafts.csswg.org/selectors/#disabled-pseudo
https://drafts.csswg.org/selectors/#required-pseudo
https://drafts.csswg.org/selectors/#optional-pseudo
https://drafts.csswg.org/selectors/#valid-pseudo
https://drafts.csswg.org/selectors/#invalid-pseudo
https://drafts.csswg.org/selectors/#user-valid-pseudo
https://drafts.csswg.org/selectors/#user-invalid-pseudo
[...]

Having very specific design rules, it feels a bit off and somehow degrades readability:
```css
input:enabled:checked { ... }
input:enabled:checked:hover { ... }
input:enabled:not(:checked) { ... }
input:disabled:checked { ... }
input:disabled:not(:checked) { ... }
```

**So for convenience: why don't we have an `:unchecked` ?**

**Comment by LcsGa** [comment]
Whereas I get the idea, then we could add `:unhover`, :unfocus`, ect. which feels weird in my personal opinion.

**Comment by SebastianZ** [comment]
The difference to `:hover` or `:focus` is that the checked state is something defined within HTML, like the disabled or optional state. And it is also part of form elements for which we already have the negative variants, so an `:unchecked` pseudo-class aligns with them.

Sebastian

**Comment by LcsGa** [comment]
I didn't see things that way! You're right! 😁

**Comment by JoshTumath** [comment]
Agreed. Authors will probably be expecting an `:unchecked` pseudo-class, and I imagine that it wouldn't be too much effort for user agents to implement support for this. An easy win.

**Comment by bluemoehre** [comment]
So, what's the next step? Should I bring up a PR? (I'm kinda afraid - haha 😄)

**Comment by SebastianZ** [comment]
As this proposal seems reasonable, the next step is for the CSS Working Group (CSSWG) to discuss it and resolve on whether to add this feature or not. Therefore, I marked this issue with the https://github.com/w3c/csswg-drafts/labels/Agenda%2B label.

@bluemoehre If the working group decides to add this feature, you are welcome to provide a PR for it. Members of the working group will guide you.

Sebastian

**Comment by LeaVerou** [comment]
On one hand, we do have both `:enabled` and `:disabled`, `:valid` and `:invalid`. A crucial point is that these are not equivalent to the `:not()` versions of the positive pseudo-class. `:disabled` is not simply `:not(:enabled)` as the latter would also match elements that cannot be activated. Similarly, `:invalid` is not simply `:not(:valid)`.
I think `:unchecked` could work similarly, and only target checkboxes and radios.

**Comment by tabatkins** [comment]
I won't be around for this discussion, so just commenting here: sounds fine to me? The point that we have a bunch of boolean pseudo pairs does suggest a pattern we should maintain for author convenience. We *can* target the same set of elements using `:not()`, since the set of elements `:checked` can apply to are very limited in the first place; `input[type="checkbox"]:not(:checked)` (or `[type="radio"]`), but granted that's pretty long. ^_^

I was going to say something about `ElementInternals` and form-associated custom elements, but it seems that that API doesn't currently have a way to associate an element with `:checked` anyway. That might be something we should think about in the future; if we do add such an API, it will need to be able to distinguish between checked/unchecked and not checkable. (It'll already need to be a multi-state concept anyway, to handle `:indeterminate`.) Anyway, that's a tangent and not germane to this issue.

**Comment by mayank99** [comment]
A small consideration that may have been missed: In HTML/JS, `indeterminate` and `checked` states are _not_ mutually exclusive; which means `:indeterminate` can be made to match together with both `:checked` and `:not(:checked)`. It may not be very useful, but you can have "indeterminate checked checkbox" and "indeterminate unchecked checkbox".

<details><summary>You can verify it in the browser console</summary>

```js
const checkbox = document.createElement("input");
checkbox.type = "checkbox";
checkbox.checked = true;
checkbox.indeterminate = true;

checkbox.matches(":checked"); // true
checkbox.matches(":indeterminate"); // true
```

</details> 

If `:unchecked` is made mutually exclusive with `:indeterminate`, it would make things inconsistent.

**Comment by bluemoehre** [comment]
Interesting point - while browsers can visualize the indeterminate state on a standalone checkbox, it is not accessible to users directly. In my understanding of the specs the `indeterminate` state is just a visualization controlled by JS and not linked to the un-/checked state in any way.

In my logical understanding an input is always unchecked if ...

`input[type=checkbox|radio]`
- initially not checked OR
- unchecked by user

Did I miss something?

**Comment by tabatkins** [comment]
Re: indeterminate and checked/uncheckedness, this was already the case. The CSS :checked and :indeterminate pseudo-classes were *defined* to be mutually exclusive, and browsers obey this, even tho a checkbox can be both `checked=true` and `indeterminate=true` (and thus will submit its value in a form); in this case, it matches :indeterminate but not :checked. The :unchecked language is just following along in this regard.

However, I have made it clearer in the spec exactly what the interaction between indeterminate and checked/unchecked should be when they're handled by separate booleans in the host language.
